lib/open_trip_planner_client/error.ex

```diff
@@ -0,0 +1,97 @@
+defmodule OpenTripPlannerClient.Error do
+  @moduledoc """
+  Describes errors from OpenTripPlanner, including routing errors from the /plan
+  endpoint for now. For routing errors, generates custom human-friendly error
+  messages based on routing error code and plan details.
+  """
+
+  alias OpenTripPlannerClient.Plan
+  alias Timex.{Duration, Format.Duration.Formatter}
+
+  require Logger
+
+  defstruct [:details, :message, :type]
+
+  @type t :: %__MODULE__{
+          details: map(),
+          message: String.t(),
+          type: :graphql_error | :routing_error
+        }
+
+  @spec from_graphql_error(map()) :: t()
+  def from_graphql_error(error) do
+    _ = log_error(error)
+
+    {message, details} = Map.pop(error, :message)
+    %__MODULE__{details: details, message: message, type: :graphql_error}
+  end
+
+  @spec from_routing_errors(Plan.t()) :: t()
+  def from_routing_errors(%Plan{routing_errors: routing_errors} = plan) do
+    _ = log_error(routing_errors)
+
+    for %{code: code, description: description} <- routing_errors do
+      %__MODULE__{
+        message: code_to_message(code, description, plan),
```

Comment:
suggestion: I think it would be worth preserving this code as well to support localization on the client side.

Reply:
I agree with this. Also, that way if we wind up having to use the fallback error message, it's easier to debug why.

Reply:
Can you expand on this a little? I've too little experience with localization but I want to better understand what doing this actually entails!

Reply:
Sure thing! Since the mobile app needs to have manually added translations for every string that a user will see (or hear for voice over users), it is easier for us to create set of strings that need to be translated from a field with a limited set of values rather than a free-text field like `message`. 


Some ios-specific details on what that looks like in practice: We currently maintain a[ localization file ](https://github.com/mbta/mobile_app/blob/main/iosApp/iosApp/Localizable.xcstrings) that contains all the strings with their translations. Here is an example where we [turn enum values for an alert effect](https://github.com/mbta/mobile_app/blob/f510028d80a3a4172c8f5e6b68dc24a32755c782/iosApp/iosApp/Utils/AlertExtension.swift#L14) returned from the backend into translatable text strings (the `Text` view does some magic in XCode to automatically create placeholder entries in the localization file, which we then can manually add or import translations for). 


In the future if we know that we definitely want to display the same error strings across web and mobile we could  consider passing a locale code to the trip planning endpoint and push more localization server-side. It looks like [OTP supports passing a locale code via the accept-language](https://github.com/opentripplanner/OpenTripPlanner/pull/1917), which we may also want to explore.